Warnings in builing lib on GCC8

<!-- Before filling this issue, please read the wiki (https://github.com/fossology/fossology/wiki)
and search if the bug do not already exists in the issues (https://github.com/fossology/fossology/issues). -->

### Description

While builidling FOSSology lib using GCC 8.3.0 on Debian 10, I recieve following warnings.

#### How to reproduce

Run `make build-lib` with gcc 8.

### Logs

```lisp
libfossdb.c: In function ‘fo_dbconnect’:
libfossdb.c:150:75: warning: ‘%s’ directive output may be truncated writing up to 10239 bytes into a region of size 960 [-Wformat-truncation=]
         "ERROR: Unable to connect to the database\n   Connection string: '%s'\n   Connection status: '%d'\n  Check: /usr/local/etc/fossology/Db.conf\n", CMD, PQstatus(pgConn));
                                                                           ^~                                                                             ~~~
libfossdb.c:150:9: note: using the range [-2147483648, 2147483647] for directive argument
         "ERROR: Unable to connect to the database\n   Connection string: '%s'\n   Connection status: '%d'\n  Check: /usr/local/etc/fossology/Db.conf\n", CMD, PQstatus(pgConn));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libfossdb.c:149:7: note: ‘snprintf’ output between 135 and 10384 bytes into a destination of size 1024
       snprintf(*ErrorBuf, ERRBUFSIZE,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "ERROR: Unable to connect to the database\n   Connection string: '%s'\n   Connection status: '%d'\n  Check: /usr/local/etc/fossology/Db.conf\n", CMD, PQstatus(pgConn));
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libfossrepo.c: In function ‘fo_RepOpenFull’:
libfossrepo.c:974:3: warning: ‘strncpy’ specified bound 1025 equals destination size [-Wstringop-truncation]
   strncpy(RepPath, path, sizeof(RepPath));
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -c sqlCopy.c -DFOSSDB_CONF='"/usr/local/etc/fossology/Db.conf"' -DFOSSGROUP='"fossy"' -DDEFAULT_SETUP='"/usr/local/etc/fossology"' -fPIC -I/usr/include/postgresql -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I../../../src/lib/c -g -O2 -Wall -D_FILE_OFFSET_BITS=64
sqlCopy.c: In function ‘fo_sqlCopyCreate’:
sqlCopy.c:113:24: warning: ‘strncat’ specified bound 1 equals source length [-Wstringop-overflow=]
       if (ColIdx != 0) strncat(pCopy->ColumnNames, ",", 1);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/lib/c/libfossdb.c
+++ b/src/lib/c/libfossdb.c
@@ -20,7 +20,7 @@ along with this library; if not, write to the Free Software Foundation, Inc.0
  \brief Common libpq database functions.
  */
 
-#define ERRBUFSIZE 1024
+#define ERRBUFSIZE 11264
 
 #include "libfossdb.h"
 

--- a/src/lib/c/libfossrepo.c
+++ b/src/lib/c/libfossrepo.c
@@ -971,7 +971,8 @@ int fo_RepOpenFull(fo_conf* config)
     fprintf(stderr, "ERROR %s.%d: %s\n", __FILE__, __LINE__, error->message);
     return 0;
   }
-  strncpy(RepPath, path, sizeof(RepPath));
+  strncpy(RepPath, path, sizeof(RepPath)-1);
+  RepPath[sizeof(RepPath)-1] = 0;
 
   return 1;
 } /* fo_RepOpen() */

